engine/grid.py

```diff
@@ -435,26 +524,24 @@ def get_border_lawnmower_waypoints(self):
         """
         waypoints = []
         node_list = self.nodes
```

Comment:
does node_list only contain the active nodes (nodes that we will traverse in lawnmower) or all nodes (including the blue inactive nodes)?

Reply:
it looks like it is the entire grid. It is defined like this
`    self.nodes = generate_nodes(
            lat_min, long_min, self.num_rows, self.num_cols, STEP_SIZE_METERS)`
`generate_nodes Returns a 2D Numpy array of Node objects that make up the entire grid. Has dimensions [rows x cols]`

Reply:
do you know why we only check the 0th col and the self.num_rows-1 col for get_border_lawnmower_waypoints?